### PR TITLE
Release notes

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-spark-connector/blob/master/REVIEWING.md)

Staging - https://docs-mongodbcom-staging.corp.mongodb.com/spark-connector/docsworker-xlarge/release-notes/release-notes/

## Self-Review Checklist

- [X] Is this free of any warnings or errors in the RST?
- [X] Did you run a spell-check?
- [X] Did you run a grammar-check?
- [X] Are all the links working?

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -4,9 +4,22 @@ Release Notes
...
 
-*Released on March 31, 2022*
+- Support for Scala 2.13.
+- Support for micro-batch mode with Spark Structured Streaming.
+- Support for BSON data types.
+- Improved partitioner support for empty collections.
+- Option to disable automatic upsert on write operations.
+- Improved schema inference for empty arrays.
+- Support for null values in arrays and lists. The Connector now stores these values
+  to MongoDB instead of throwing an exception.
```
Comment (DBirtolo-mdb) [inline source/release-notes.txt:17]: [suggestion] I feel like the preposition doesn't match the verb here. Maybe have it be "...writes these values to MongoDB..." or "...stores these values in MongoDB...".

File: source/release-notes.txt [hunk 2]
```diff
@@ -4,9 +4,22 @@ Release Notes
...
+
+See `this post <https://www.mongodb.com/blog/post/introducing-mongodb-spark-connector-version-10-1>`__ 
+on the MongoDB blog for more information.
+
+MongoDB Connector for Spark 10.0.0
+----------------------------------
 
-- Support for Spark Structured Streaming.
+- Support for Spark Structured Streaming
```
Comment (DBirtolo-mdb) [inline source/release-notes.txt:25]: [nitpick] The period should still be at the end of this item.

File: source/release-notes.txt [hunk 3]
```diff
@@ -4,9 +4,22 @@ Release Notes
 
 .. default-domain:: mongodb
 
-MongoDB Connector for Spark 10.0.0
+MongoDB Connector for Spark 10.1.0
 ----------------------------------
 
-*Released on March 31, 2022*
```
Comment (DBirtolo-mdb) [inline source/release-notes.txt:10]: [question] I'm not sure of MongoDB standards, should we include release dates?

**Review by DBirtolo-mdb — CHANGES_REQUESTED** [review]
Good work - just a couple of minor things.

File: source/release-notes.txt (hunk 3 above)
Comment (mongoKart) [reply source/release-notes.txt:10]: I was taking cues from [our other release notes](https://www.mongodb.com/docs/drivers/node/current/whats-new/#std-label-version-4.13), which don't include dates.

**Review by DBirtolo-mdb — APPROVED** [review]
LGTM!

**Pull request merged**